libia2/ia2.c

```diff
@@ -132,6 +134,37 @@ int protect_pages(struct dl_phdr_info *info, size_t size, void *data) {
     break;
   }
 
+  /* protect TLS segment */
+  for (size_t i = 0; i < info->dlpi_phnum; i++) {
+    Elf64_Phdr phdr = info->dlpi_phdr[i];
+    if (phdr.p_type != PT_TLS) {
+      continue;
+    }
+
+    void *start = info->dlpi_tls_data;
+    size_t len = phdr.p_memsz;
+
+    void *start_round_down = (void *)((uint64_t)start & ~0xFFF);
```

Comment:
`~0xFFFULL` to avoid integer casting issues? Or maybe use `PAGE_SIZE - 1` for more clarity and portability?

Reply:
The rest of the file uses [`~`]`0xFFFUL`; I'll make this match.